spanner/api/Spanner/Program.cs

```diff
@@ -1343,6 +1378,177 @@ await connection.CreateDdlCommand($"DROP TABLE {table}")
+                        {"SingerId", SpannerDbType.String},
+                        {"FirstName", SpannerDbType.String},
+                        {"LastName", SpannerDbType.String}
+                    }))
+                    {
+                        cmd.Transaction = tx;
+                        for (var x = 1; x < 2500; x++)
+                        {
+                            maxSingerId++;
+                            string nameSuffix = Guid.NewGuid().ToString().Substring(0, 8);
+                            cmd.Parameters["SingerId"].Value = maxSingerId;
+                            cmd.Parameters["FirstName"].Value = $"FirstName-{nameSuffix}";
+                            cmd.Parameters["LastName"].Value = $"LastName-{nameSuffix}";
+                            cmd.ExecuteNonQuery();
+                        }
+                        await tx.CommitAsync();
+                    }
+                }
+            }
+            Console.WriteLine("Done inserting sample records...");
+        }
+
+        // [START spanner_batch_client]
+        public static object BatchReadRecords(string projectId,
+             string instanceId, string databaseId)
+        {
+            var response =
+                DistributedReadAsync(projectId, instanceId, databaseId);
+            Console.WriteLine("Waiting for operation to complete...");
+            response.Wait();
+            Console.WriteLine($"Operation status: {response.Status}");
+            return ExitCode.Success;
+        }
+
+        private static async Task DistributedReadAsync(string projectId,
+            string instanceId, string databaseId)
+        {
+            string connectionString =
+                $"Data Source=projects/{projectId}/instances/{instanceId}"
+                + $"/databases/{databaseId}";
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                await connection.OpenAsync();
+
+                using (var transaction =
+                    await connection.BeginReadOnlyTransactionAsync())
+                using (var cmd =
+                    connection.CreateSelectCommand(
+                        "SELECT SingerId, FirstName, LastName FROM Singers"))
+                {
+                    transaction.DisposeBehavior =
+                        DisposeBehavior.CloseResources;
+                    cmd.Transaction = transaction;
+                    var partitions = await cmd.GetReaderPartitionsAsync();
+                    var transactionId = transaction.TransactionId;
+                    await Task.WhenAll(partitions.Select(
+                            x => DistributedReadWorkerAsync(x, transactionId)))
+                                .ConfigureAwait(false);
+                }
+                Console.WriteLine($"Done reading!  Total rows read: "
+                    + $"{s_rowsRead:N0} with {s_partitionId} partition(s)");
+            }
+        }
+
+        private static int s_partitionId;
```

Comment:
nit: All the static variables should be defined before all the methods

Reply:
Moved the static variables to the top of the code snippet utilizing them so they remain included in the code snippet to which they are relevant.